services/reporting/reporting.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+package reporting
+
+import (
+	"time"
+
+	"github.com/getsentry/sentry-go"
+)
+
+// Return the options that can be configured for the error reporting
+// library.
+func Options() sentry.ClientOptions {
```

Comment:
I have my doubts about this `Options` function here, I don't think is needed.

Reply:
I added because of this:

Imagine another module that depends on reporting, the way it's done now it would work like this:

```go
// module X
import "...services/reporting"

options := Options()
options.Dsn = "..."
```

Now, without the `Options()` function it will look like:

```go
// module X
import "...services/reporting"
import "...sentry-go"          // direct dependency to sentry-go

options := ClientOptions()  // it comes from sentry-go
options.Dsn = "..."
```

So that is why I added it, to avoid a direct dependency to `sentry-go` and have somewhat control of how options are created if needed for some reason.

Reply:
From my perspective, the reporting service must be a black box from the outside, if I have to work with sentry option objects, I'm leaking that part of the implementation, and can generate problems if in the future we replace it with another reporting service. I prefer to generate our own Options object, that receive only the options needed from the outside, and then generate the corresponding sentry options object.